includes/class-wc-subscriptions-core-plugin.php

```diff
@@ -326,19 +327,6 @@ public function get_gateways_handler_class() {
 	 * @since 1.0.0 - Migrated from WooCommerce Subscriptions v4.0.0
 	 */
 	public function register_order_types() {
-		$subscriptions_exist = $this->cache->cache_and_get( 'wcs_do_subscriptions_exist', 'wcs_do_subscriptions_exist' );
-
-		if ( true === (bool) apply_filters( 'woocommerce_subscriptions_not_empty', $subscriptions_exist ) ) {
-			$not_found_text = __( 'No Subscriptions found', 'woocommerce-subscriptions' );
-		} else {
-			$not_found_text = '<p>' . __( 'Subscriptions will appear here for you to view and manage once purchased by a customer.', 'woocommerce-subscriptions' ) . '</p>';
-			// translators: placeholders are opening and closing link tags
-			$not_found_text .= '<p>' . sprintf( __( '%1$sLearn more about managing subscriptions &raquo;%2$s', 'woocommerce-subscriptions' ), '<a href="http://docs.woocommerce.com/document/subscriptions/store-manager-guide/#section-3" target="_blank">', '</a>' ) . '</p>';
-			// translators: placeholders are opening and closing link tags
-			$not_found_text .= '<p>' . sprintf( __( '%1$sAdd a subscription product &raquo;%2$s', 'woocommerce-subscriptions' ), '<a href="' . esc_url( WC_Subscriptions_Admin::add_subscription_url() ) . '">', '</a>' ) . '</p>';
-		}
-
-		$subscriptions_not_found_text = apply_filters( 'woocommerce_subscriptions_not_found_label', $not_found_text );
```

Comment:
All this logic was kinda hard to understand tbh. It looks as though there's a filter in here that I'll need to honour though. 

If subscriptions do exists (`$subscriptions_exist === true`) or a third-party says they exists, but WP doesn't load anything into the table it use to say "No subscriptions found".

<img width="518" alt="Screenshot 2023-08-02 at 10 50 22 am" src="https://github.com/Automattic/woocommerce-subscriptions-core/assets/8490476/87648fc4-4d42-4771-bcbd-4e81d93f4f90">

Whereas if there are no subscriptions or something doesn't change the `$subscriptions_exist` value via the filter, then it displays this:

<img width="814" alt="Screenshot 2023-08-02 at 11 32 53 am" src="https://github.com/Automattic/woocommerce-subscriptions-core/assets/8490476/89ff2e48-3791-494b-a244-fc5051632201">

I don't know if we want to preserve this nuance at all.

Reply:
wait 🤦‍♂️. 

This is used when you use filters an no subscriptions are returned. 🤔

Reply:
I've fixed these issues in more recent commits.